commit 92c770e452c481cd1bc81bdb093733d1007602fb
Author: Aaron Sierra <asierra@xes-inc.com>
Date:   Tue Jul 24 14:23:46 2018 -0500

    serial: 8250_exar: Read INT0 from slave device, too
    
    commit 60ab0fafc4b652fcaf7cbc3bb8555a0cf1149c28 upstream.
    
    The sleep wake-up refactoring that I introduced in
    
      commit c7e1b4059075 ("tty: serial: exar: Relocate sleep wake-up handling")
    
    did not account for devices with a slave device on the expansion port.
    This patch pokes the INT0 register in the slave device, if present, in
    order to ensure that MSI interrupts don't get permanently "stuck"
    because of a sleep wake-up interrupt as described here:
    
      commit 2c0ac5b48a35 ("serial: exar: Fix stuck MSIs")
    
    This also converts an ioread8() to readb() in order to provide visual
    consistency with the MMIO-only accessors used elsewhere in the driver.
    
    Reported-by: Andy Shevchenko <andriy.shevchenko@linux.intel.com>
    Signed-off-by: Aaron Sierra <asierra@xes-inc.com>
    Fixes: c7e1b4059075 ("tty: serial: exar: Relocate sleep wake-up handling")
    Reviewed-by: Andy Shevchenko <andriy.shevchenko@linux.intel.com>
    Cc: stable <stable@vger.kernel.org>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/tty/serial/8250/8250_exar.c b/drivers/tty/serial/8250/8250_exar.c
index 38af306ca0e8..a951511f04cf 100644
--- a/drivers/tty/serial/8250/8250_exar.c
+++ b/drivers/tty/serial/8250/8250_exar.c
@@ -433,7 +433,11 @@ static irqreturn_t exar_misc_handler(int irq, void *data)
 	struct exar8250 *priv = data;
 
 	/* Clear all PCI interrupts by reading INT0. No effect on IIR */
-	ioread8(priv->virt + UART_EXAR_INT0);
+	readb(priv->virt + UART_EXAR_INT0);
+
+	/* Clear INT0 for Expansion Interface slave ports, too */
+	if (priv->board->num_ports > 8)
+		readb(priv->virt + 0x2000 + UART_EXAR_INT0);
 
 	return IRQ_HANDLED;
 }
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1030_92c770e452c481cd1bc81bdb093733d1007602fb.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1030_92c770e452c481cd1bc81bdb093733d1007602fb.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
